python/ray/data/aggregate/_aggregate.py

```diff
@@ -41,18 +43,16 @@ class AggregateFn:
             accumulate_block must be provided.
         finalize: This is called once to compute the final aggregation
             result from the fully merged accumulator.
-        name: The name of the aggregation. This will be used as the output
-            column name in the case of Arrow dataset.
     """
 
     def __init__(
         self,
         init: Callable[[KeyType], AggType],
         merge: Callable[[AggType, AggType], AggType],
+        name: str,
         accumulate_row: Callable[[AggType, T], AggType] = None,
         accumulate_block: Callable[[AggType, Block], AggType] = None,
         finalize: Callable[[AggType], U] = lambda a: a,
-        name: Optional[str] = None,
```

Comment:
Should also add a check. Changing the type hint doesn't have any effect at runtime.